backend/app/services/otp_service.py

```
@@ -0,0 +1,100 @@
+        return f"{ttl_seconds} {unit}"
+
+    minutes = (ttl_seconds + 59) // 60
+    unit = "minute" if minutes == 1 else "minutes"
+    return f"{minutes} {unit}"
+
+
+def request_otp(settings: Settings, farmer_repo: FarmerRepository, farmer_id: str) -> OtpRequestOut:
+    """Generate and send a phone-verification OTP to the farmer's registered number."""
+    farmer = farmer_repo.get(farmer_id)
+    if farmer is None:
+        raise NotFoundError("Farmer profile not found - register first")
```

Ruff (0.16.3)

[warning] Avoid specifying long messages outside the exception class

(TRY003)